scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -119,6 +119,67 @@ object TypedPipe extends Serializable {
       override def hash(x: Int): Int = x
     }
   }
+  implicit class HigherPriorityTypedPipeMethods[T, K, V](val tp: TypedPipe[T])(implicit ev: TypedPipe[T] <:< TypedPipe[(K, V)],
```

Comment:
can we just do `SketchJoinMethods` similar to https://github.com/twitter/scalding/blob/ianoc/addSketchOrderedSerializationInterface/scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala#L92 so we can fix `(K, V)` and simpify a bit